xbmc/URL.cpp

```diff
@@ -253,35 +253,29 @@ void CURL::Parse(const std::string& strURL1)
     }
   }
 
+  std::string strHostNameAndPort = strURL.substr(iPos, (iSlash == std::string::npos) ? iEnd - iPos : iSlash - iPos);
+  // check for IPv6 numerical representation inside []
+  size_t iBrk = strHostNameAndPort.rfind("]");
+  if (iBrk < iEnd && iBrk != std::string::npos && strHostNameAndPort.find("[") == 0)
+  {
+    m_strHostName = strHostNameAndPort.substr(1, iBrk-1);
+    strHostNameAndPort.erase(0, iBrk+1);
+  }
+
   // detect hostname:port/
-  if (iSlash == std::string::npos)
+  size_t iColon = strHostNameAndPort.rfind(":");
+  if (iColon != std::string::npos && iColon == strHostNameAndPort.find(":"))
```

Comment:
i was missing the fact that everything behind the slash was already stripped off in this stage... fine as is.